tithe.lic

```diff
@@ -0,0 +1,66 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#tithe
+=end
+
+custom_require.call(%w[common common-money common-travel events drinfomon])
+
+class TitheActions
+  include DRC
+  include DRCT
+  include DRCM
+
+  def initialize
+    arg_definitions = [[]]
+
+    args = parse_args(arg_definitions, true)
+
+    @settings = get_settings(args.flex)
+    @hometown = @settings.hometown
+    @pray_to_chadatru = @settings.pray_to_chadatru
+    @chadatru_prayer_cooldown = 3600
+    UserVars.chadatru_prayer_last ||= Time.now - @chadatru_prayer_cooldown
+
+    check_tithe
+    check_chadatru
+  end
+
+  def check_tithe
+    return unless @settings.tithe
+    return unless almsbox = @settings.tithe_almsbox || get_data('town')[@hometown]['almsbox']['id']
+    delta = Time.now - (UserVars.tithe_timer || Time.now - 14_400)
+    return if (delta < 4 * 60 * 60 && DRStats.paladin?) || (delta < 60 * 10 && DRStats.cleric?)
+
+    currency = hometown_currency(@hometown)
+    return unless withdraw_exact_amount?('5 silver', @settings)
+
+    wait_for_script_to_complete('pay-debt')
+
+    UserVars.tithe_timer = Time.now
+    walk_to(almsbox)
```

Comment:
💡 My preference would be to use the module prefix (e.g. `DRCT` here) whenever possible